internal/notifier/notifier.go

```diff
@@ -135,18 +136,7 @@ func (n *Notifier) getConnAndRun(ctx context.Context) {
 
 	conn, err := n.establishConn(ctx)
 	if err != nil {
-		if errors.Is(err, context.Canceled) {
```

Comment:
One thing I wasn't quite able to figure out is why this condition wasn't detecting the cancellation previously. I would've thought that even without a special cancellation error, this still would've been enough to intercept a problem before falling through to logging.

Reply:
Hmm, hard to tell without digging into the code, but maybe it's possible pgx doesn't surface this source error correctly? `establishConn` is a simple method that just returns the result of `pgx.ConnectConfig`

Reply:
Ah yeah, that's probably it actually. The error being checked is now coming from a different place (context), which would explain why this version works.